### PR TITLE
Game Failure Screen

### PR DESCRIPTION
**Pull request by mlytvynchuk**
Styled game failure screen and updated its navigation.

### CODE REVIEW

File: app/src/main/java/com/google/droidjump/DroidStartView.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Override
+    protected void onDraw(Canvas canvas) {
+        super.onDraw(canvas);
+        Bitmap fullDroidPic = decodeResource(getResources(), R.mipmap.droid);
+        int droidCountOnFullDroidPic = 9;
+        int droidStep = (int) fullDroidPic.getWidth() / droidCountOnFullDroidPic;
+        int droidCharacterStandingNumber = 4;
+        Bitmap droid = createBitmap(fullDroidPic, /* 4th droid on full droid picture */droidStep * droidCharacterStandingNumber, 0, droidStep,
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/DroidStartView.java:39]: add extra space between */ and droidStep

Comment (redls) [inline app/src/main/java/com/google/droidjump/DroidStartView.java:39]: /* 4th droid on full droid picture */ -- this comment doesn't really work as a param. Usually when we do /* ... */ it's the param named in the method definition. 

Can we add a comment before this line and explain the what you do with droidStep * droidCharacterStandingNumber?

File: app/src/main/java/com/google/droidjump/DroidStartView.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                fullDroidPic.getHeight());
+        canvas.drawBitmap(droid, 0, getHeight() - droid.getHeight(), /* paint= */null);
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/DroidStartView.java:41]: add space between param and null

File: app/src/main/java/com/google/droidjump/GameFailureFragment.java [hunk 3]
```diff
@@ -24,21 +24,32 @@
...
 import android.widget.ImageButton;
+import android.widget.LinearLayout;
 
 public class GameFailureFragment extends Fragment {
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
-        View rootView = inflater.inflate(R.layout.game_failure_screen, container, false);
+
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameFailureFragment.java:34]: remove empty line

File: app/src/main/java/com/google/droidjump/GameFailureFragment.java [hunk 4]
```diff
@@ -24,21 +24,32 @@
...
+        View rootView = inflater.inflate(R.layout.game_failure_screen, container, /* attachToRoot= */false);
         FloatingActionButton retryButton = rootView.findViewById(R.id.retry_button);
         retryButton.setOnClickListener(view -> {
             findNavController(view).navigate(R.id.action_game_failure_screen_to_game_screen);
         });
+        // Adding redirecting on click to start screen
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameFailureFragment.java:40]: nit: maybe add a new line before each of these comments?

File: app/src/main/java/com/google/droidjump/GameSuccessFragment.java [hunk 5]
```diff
@@ -24,21 +24,33 @@
...
             findNavController(view).navigate(R.id.action_game_success_screen_to_game_screen);
         });
+        // Adding redirect to start screen
         ImageButton menuButton = rootView.findViewById(R.id.success_menu_button);
         menuButton.setOnClickListener(view -> {
             findNavController(view).navigate(R.id.action_game_success_screen_to_start_screen);
         });
+        // Drawing a a droid
+        LinearLayout drawLayout = rootView.findViewById(R.id.droid_draw_view);
+        drawLayout.addView(new DroidStartView(this.getActivity()));
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameSuccessFragment.java:47]: do you need "this"?

File: app/src/main/java/com/google/droidjump/DroidStartView.java (hunk 1 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/DroidStartView.java:39]: Rewrote this part of code

File: app/src/main/java/com/google/droidjump/GameSuccessFragment.java (hunk 5 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/GameSuccessFragment.java:47]: Removed.

File: app/src/main/java/com/google/droidjump/GameFailureFragment.java [hunk 6]
```diff
@@ -24,21 +24,39 @@
...
         ImageButton menuButton = rootView.findViewById(R.id.success_menu_button);
         menuButton.setOnClickListener(view -> {
             findNavController(view).navigate(R.id.action_game_failure_screen_to_start_screen);
         });
+
+        // Drawing a a droid
+        LinearLayout drawLayout = rootView.findViewById(R.id.droid_draw_view);
+        drawLayout.addView(new DroidStartView(getActivity()));
+
+        // Adding redirecting on click to howToPlay screen
```
[the remark this answers is not in the document]
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/GameFailureFragment.java:54]: Thanks

File: app/src/main/java/com/google/droidjump/GameFailureFragment.java (hunk 3 above)
Comment (redls) [reply app/src/main/java/com/google/droidjump/GameFailureFragment.java:34]: I still see the line.

File: app/src/main/java/com/google/droidjump/GameSuccessFragment.java [hunk 7]
```diff
@@ -30,22 +30,27 @@
...
-        ImageButton menuButton = rootView.findViewById(R.id.success_menu_button);
+
+        // Redirecting to start screen
+        ImageButton menuButton = rootView.findViewById(R.id.menu_button);
         menuButton.setOnClickListener(view -> {
             findNavController(view).navigate(R.id.action_game_success_screen_to_start_screen);
         });
-        // Drawing a a droid
+
+        // Drawing of droid
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameSuccessFragment.java:49]: nit: drawing droid.

File: app/src/main/java/com/google/droidjump/GameSuccessFragment.java [hunk 8]
```diff
@@ -30,22 +30,27 @@
...
         LinearLayout drawLayout = rootView.findViewById(R.id.droid_draw_view);
-        drawLayout.addView(new DroidStartView(this.getActivity()));
-        // Adding redirect to howToPlay screen
+        drawLayout.addView(new DroidStartView(getActivity()));
+
+        // Redirecting to howToPlay screen
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/GameSuccessFragment.java:53]: nit: Redirecting to How To Play screen?

File: app/src/main/res/values/styles.xml [hunk 9]
```diff
@@ -34,4 +34,70 @@
 
     <style name="AppTheme.PopupOverlay" parent="ThemeOverlay.AppCompat.Light" />
 
+    <style name="ScreenHeader">
+        <item name="android:textColor">@android:color/black</item>
+        <item name="android:layout_marginLeft">@dimen/fab_margin</item>
+        <item name="android:layout_marginTop">@dimen/fab_margin</item>
```
Comment (redls) [inline app/src/main/res/values/styles.xml:40]: these seem to be used very often. Maybe make another style for it?

Comment (mlytvynchuk) [reply app/src/main/res/values/styles.xml:40]: A lot of my styles inherit from others. I can't inherit from multiple styles so it's probably not the case.

**Pull request merged**